src/worker/session.cpp

```diff
@@ -306,12 +306,25 @@ void worker_session_t::process_invoke(std::map<std::uint64_t, std::shared_ptr<sh
         std::make_shared<basic_receiver_t<worker_session_t>>(id, shared_from_this(), state)
     );
 
-    trace_t trace(
-        message.get_header<hpack::headers::trace_id<>>()->get_value().convert<uint64_t>(),
-        message.get_header<hpack::headers::span_id<>>()->get_value().convert<uint64_t>(),
-        message.get_header<hpack::headers::parent_id<>>()->get_value().convert<uint64_t>(),
-        event
-    );
+    auto trace_id = message.get_header<hpack::headers::trace_id<>>();
+    auto span_id = message.get_header<hpack::headers::span_id<>>();
+    auto parent_id = message.get_header<hpack::headers::parent_id<>>();
+
+    boost::optional<trace_t> trace;
+    if(trace_id && span_id && parent_id &&
+            !trace_id->value().empty() && !span_id->value().empty() && ! parent_id->value().empty()) {
+        try {
+            trace = trace_t(
+                    hpack::header::binary_unpack<uint64_t>(message.get_header<hpack::headers::trace_id<>>()->value()),
+                    hpack::header::binary_unpack<uint64_t>(message.get_header<hpack::headers::span_id<>>()->value()),
+                    hpack::header::binary_unpack<uint64_t>(message.get_header<hpack::headers::parent_id<>>()->value()),
+                    event);
+        } catch (const std::exception& e) {
+            throw;
+            CF_DBG("could not decode tracing headers - %s", e.what());
```

Comment:
On the first look simple 'printf' would be fine. Much simpler than using BH, ok for trivial debug.